src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,67 @@
+package duke.command;
+
+import java.util.Arrays;
+
+/**
+ * A Class that extends the Command class.
+ * It is specifically designed for a Command for adding.
+ *
+ * @author Gu Geng
+ */
+public class AddCommand extends Command {
+    private duke.task.Task task;
+
+    /**
+     * Returns a AddCommand object with the information provided.
+     *
+     * @param command A String containing information that can possibility be used to create an AddCommand object.
+     * @throws duke.DukeException Will be thrown if information provided are insufficient/incorrect.
+     */
+    public AddCommand(String... command) throws duke.DukeException {
+        if (command.length == 1) {
+            throw new duke.DukeException(" ☹ OOPS!!! The description of a task cannot be empty.");
+        } else {
+            if (command[0].equals("todo")) {
+                task = new duke.task.Todo(command);
+            } else {
+                if (Arrays.stream(command).anyMatch("/"::equals) && command.length == 2) {
+                    throw new duke.DukeException(
+                            " ☹ HEY DEAR! Please enter a date after / following the task description");
+                }
+                if (command[0].equals("deadline")) {
+                    task = new duke.task.Deadline(command);
+                } else {
+                    task = new duke.task.Event(command);
+                }
+            }
+        }
+    }
```

Comment:
Perhaps it will be better formatting to use 
```
if (condition) {
    statements;
} else if (condition) {
    statements;
} else {
    statements;
}
``` 
instead of nested
```
if (condition) {
    statements;
} else {
    statements;
}
```

Reply:
You have a valid point here! Thanks!